src/codegen/compile.ml

```diff
@@ -540,6 +547,14 @@ module E = struct
     in
     let gc_fn = if !Flags.force_gc then gc_fn else "schedule_" ^ gc_fn in
     call_import env "rts" (gc_fn ^ "_gc")
+
+  (* See Note [Candid subtype checks] *)
+  let add_typtbl_typ (env : t) ty : Int32.t =
```

Comment:
```suggestion
  (* NB: we don't bother detecting duplicate registrations here because the sharing machinery should ensure
     that `add_typtbl_typ t` is called at most once for any `t`  with a distinct type hash *)
  let add_typtbl_typ (env : t) ty : Int32.t =
```